### ISSUE TITLE
Add "test against gold" feature

### ISSUE THREAD

**Issue by ChandlerSwift** [UMDLARS/dtanm#103]
We have a UI mocked out for this ([test_against_gold.html](https://github.com/UMDLARS/dtanm/blob/6835f28afeddd886a5950d475ff0a2efb037c2a5/web/web/templates/test_against_gold.html)), but it was removed from the menu in 588188538566dd8390b503d11eea7ba2f510afed since it wasn't yet functional. 

Open questions (from Slack):
@pahp: 
- In your mind, would this be simply for running `gold` to see what it does, or run it like an attack, where we print the output of both `gold` and the team's executable, and then compare them? I think the first might be _slightly_ easier, but they should be pretty similar, since we've already got all the logic worked out anyway.
- Since users probably want this for fairly quick feedback, do you think it makes sense to prioritize these runs above the "normal" runs (that is, the normal "attack vs. team code" runs). Does that make sense, or can you think of any problems that might cause?
- I figure I'd rate limit these, probably 6-ish per minute, similarly to attacks. I can't think of any issues we'd have without rate limiting other than potentially denial of service, but it seems the safer bet. And 6 per minute is fairly generous, I think.

**Comment by pahp** [comment]
Hi, sorry for the delay. This just fell out of scope but is easy to answer, I think.

- Just running gold would be awesome. Comparing against the current submitted code would be SUPER AWESOME!
- prioritizing runs makes sense
- yes, rate limiting also makes sense. I agree that 6 is generous. Six per team per minute? Once per minute per user? Really make them think about it could be interesting. Rate limiting for TAG could be a game option, I guess. Nice for newbies, lock it down for oldbies?

Thanks again for all your incredible work.

**Comment by ChandlerSwift** [comment]
The easiest way to do this seems like to extend the "attack" type to include the tests against gold. If we add a `type` field to `attacks`, this could be set to "test" in the test_against_gold case, and "attack" otherwise. Likewise for our `Result`s. 

- [ ] Add help text to Test against Gold page
- [ ] add `type` field to attacks and filter on:
	- [ ] teams/{id}
	- [ ] team ranking (teams/index) -- this one may be automatic
	- [ ] attacks
	- [ ] disallow attacks/{id} for non-attack types
	- [ ] test against gold attack list
	- [ ] competition stats page
- [ ] add `type` field to results and filter on
	- [ ] teams/{id}
	- [ ] team ranking?
- [ ] Allow to use existing attacks
- [ ] Add adding manual tests
- [ ] Allow testing against custom branch (Probably depends on #116)
- [ ] Modify worker to process new entries differently
